### ISSUE TITLE
Calls to passwords API fail with error 500 with username that includes an at (@) character.

### ISSUE THREAD

**Issue by ricalo** [daper/nextcloud-passwords-app#44]
After successfully providing credentials, the app tries to call the `/api/1.0/password/list` endpoint. The app only displays a blank screen.

Logcat displays the following:

```yaml
09-01 16:10:48.667 16560 16591 I ReactNativeJS: Fetching passwords
09-01 16:10:48.690 16560 16591 I ReactNativeJS: 'Starting Request', { adapter: [Function: xhrAdapter],
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   transformRequest: { '0': [Function: transformRequest] },
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   transformResponse: { '0': [Function: transformResponse] },
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   timeout: 120000,
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   xsrfCookieName: 'XSRF-TOKEN',
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   xsrfHeaderName: 'X-XSRF-TOKEN',
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   maxContentLength: -1,
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   validateStatus: [Function: validateStatus],
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   headers:
09-01 16:10:48.690 16560 16591 I ReactNativeJS:    { common: { Accept: 'application/json, text/plain, */*' },
09-01 16:10:48.690 16560 16591 I ReactNativeJS:      delete: {},
09-01 16:10:48.690 16560 16591 I ReactNativeJS:      get: {},
09-01 16:10:48.690 16560 16591 I ReactNativeJS:      head: {},
09-01 16:10:48.690 16560 16591 I ReactNativeJS:      post: { 'Content-Type': 'application/x-www-form-urlencoded' },
09-01 16:10:48.690 16560 16591 I ReactNativeJS:      put: { 'Content-Type': 'application/x-www-form-urlencoded' },
09-01 16:10:48.690 16560 16591 I ReactNativeJS:      patch: { 'Content-Type': 'application/x-www-form-urlencoded' },
09-01 16:10:48.690 16560 16591 I ReactNativeJS:      'OCS-APIRequest': 'true' },
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   method: 'post',
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   baseURL: 'https://nextcloud.example.com/index.php/apps/passwords',
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   auth:
09-01 16:10:48.690 16560 16591 I ReactNativeJS:    { username: 'username%40example.com',
09-01 16:10:48.690 16560 16591 I ReactNativeJS:      password: 'long_hashed_or_so_password' },
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   url: '/api/1.0/password/list',
09-01 16:10:48.690 16560 16591 I ReactNativeJS:   data: { detailLevel: 'model+folder' } }
09-01 16:11:14.169 16560 16591 I ReactNativeJS: error while getting passwords Error: Request failed with status code 500
```

Additionally, the Nextcloud server logs show the following messages:

```json
{
"reqId":"reqId",
"level":2,
"time":"2019-09-01T20:22:47+00:00",
"remoteAddr":"client_ip",
"user":"--",
"app":"core",
"method":"POST",
"url":"/index.php/apps/passwords/api/1.0/folder/list",
"message":"Login failed: 'username%40example.com' (Remote IP: 'client_ip')",
"userAgent":"okhttp/3.12.1",
"version":"16.0.4.1","id":"an_id"
}

[passwords] Warning: Login attempt with invalid authorization header for invalid user id

POST /index.php/apps/passwords/api/1.0/folder/list
from client_ip at 2019-09-01T20:22:47+00:00
```

It looks to me like the app is incorrectly encoding the at `@` sign in my username. To verify, I hardcoded my username in the [`src/API/index.js`][0] file using `username@example.com` instead of `username%40example.com`. The calls to the API work correctly and the screen loads my passwords successfully.

[0]: https://github.com/daper/nextcloud-passwords-app/blob/master/src/API/index.js#L35

**Comment by rottenIO** [comment]
I can confirm, the @ server is trying to auth usernames containing @ as if they were %40. So they are not correctly encoded

**Comment by daper** [comment]
Released on v1.13.
